src/components/AnimateSharedLayout/utils/batcher.ts

```diff
@@ -62,7 +62,21 @@ export function createBatcher(): SyncLayoutBatcher {
             flushSync.preRender()
             flushSync.render()
 
+            /**
+             * Schedule a callback at the end of the following frame to assign the latest projection
+             * box to the prevViewportBox snapshot. Once global batching is in place this could be run
+             * synchronously. But for now it ensures that if any nested `AnimateSharedLayout` top-level
+             * child attempts to calculate its previous relative position against a prevViewportBox
+             * it will be against its latest projection box instead, as the snapshot is useless beyond this
+             * render.
+             */
+            sync.postRender(() => order.forEach(assignProjectionToSnapshot))
```

Comment:
i there a test for the bug that this fixes? i'm afraid it will return in some shape or form once we do the global batching

Reply:
I'll have to revisit this with a test as I think it'll be quite difficult to implement and we want to ship this out asap